jujugui/static/gui/src/app/app.js

```diff
@@ -416,50 +413,48 @@ YUI.add('juju-gui', function(Y) {
       var ecs = new juju.EnvironmentChangeSet({db: this.db});
       ecs.on('changeSetModified', this._renderDeploymentBar.bind(this));
       ecs.on('currentCommitFinished', this._renderDeploymentBar.bind(this));
-      // Instantiate the Juju environment.
-      this._generateSocketUrl(function(socketUrl, user, password) {
-        this.set('socket_url', socketUrl);
-        var envOptions = {
-          ecs: ecs,
-          socket_url: socketUrl,
-          user: user,
-          password: password,
-          readOnly: this.get('readOnly'),
-          conn: this.get('conn'),
-          jujuCoreVersion: this.get('jujuCoreVersion')
-        };
-        var webModule = environments.web;
-        if (this.get('sandbox')) {
-          envOptions.socket_url = this.get('sandboxSocketURL');
-          // The GUI is running in sandbox mode.
-          var sandboxModule = environments.sandbox;
-          if (envOptions.user && envOptions.password) {
-            var credentials = state.get('authorizedUsers');
-            credentials['user-' + envOptions.user] = envOptions.password;
-            state.set('authorizedUsers', credentials);
-          }
-          envOptions.conn = new sandboxModule.ClientConnection({
-            juju: new sandboxModule.GoJujuAPI({
-              state: state,
-              socket_url: envOptions.socket_url
-            })
-          });
-          // Instantiate a fake Web handler, which simulates the
-          // request/response communication between the GUI and the juju-core
-          // HTTPS API.
-          envOptions.webHandler = new webModule.WebSandbox({state: state});
-        } else {
-          // The GUI is connected to a real Juju environment.
-          // Instantiate a Web handler allowing to perform asynchronous HTTPS
-          // requests to the juju-core API.
-          envOptions.webHandler = new webModule.WebHandler();
+
+      let envOptions = {
+        ecs: ecs,
+        user: this.get('user'),
+        password: this.get('password'),
+        readOnly: this.get('readOnly'),
+        conn: this.get('conn'),
+        jujuCoreVersion: this.get('jujuCoreVersion')
+      };
+
+      const jujuEnvUUID = window.juju_config && window.juju_config.jujuEnvUUID;
+      if (this.get('sandbox')) {
+        envOptions = this.createSandboxConnection(envOptions);
+      } else {
+        if (jujuEnvUUID) {
+          // If we have a uuid provided in the config then connect directly
+          // to that instead of waiting for the model list from the controller
+          // connection.
+          const socketURL = this.createSocketURL(jujuEnvUUID);
+          envOptions.socket_url = socketURL;
+          this.set('socket_url', socketURL);
```

Comment:
currently envOptions is shared between the two api instances but they will be split into separate objects in the near future. Also this sets the socket_url in the app whereas the envOptions is a configuration object used to instantiate the API instances so both would be required anyways.